src/client/mod.rs

```diff
@@ -309,6 +310,27 @@ impl Client {
         pipeline: impl IntoIterator<Item = Document>,
         options: Option<ChangeStreamOptions>,
     ) -> Result<ChangeStream> {
-        todo!();
+        let pipeline: Vec<Document> = pipeline.into_iter().collect();
+        let mut options = options;
+        resolve_options!(self, options, [selection_criteria]);
+
+        let watch = Watch::new(
+            ChangeStreamTarget::Cluster(self.database("admin")),
+            pipeline.clone(),
+            options.clone(),
+        )?;
+        let cursor = self
+            .execute_cursor_operation(watch)
+            .await
+            .map(|(spec, session)| Cursor::new(self.clone(), spec.cursor_spec, session))?;
```

Comment:
I'm a bit confused by what's being done here; `execute_cursor_operation` is being called with a `Watch` operation and returns a `ChangeStreamSpecification`, which contains the pipeline, the options, and the cursor spec, but then the `CursorSpecification` is being taken out and the `ChangeStreamSpecification` is thrown away. As it's written right now, the `ChangeStreamSpecification` type isn't being used for anything. I think it would make more sense to have `ChangeStream::new` take in a `ChangeStreamSpecification` the same way that `Cursor::new` takes in a `CursorSpecification`, and then the `ChangeStream` can construct the `Cursor` itself.

Reply:
fixed